### PR TITLE
Update Cluster Setup documentation

### PR DESCRIPTION
**Pull request by cdrage**
**What type of PR is this:**

/kind documentation

**What does this PR do / why we need it:**

Updates the documentation for cluster setup with the following:
- Removes trailing sentences / reduces the amount of uneeded sentences
- Organizes the information logically on both k8s and oc pages
- Changes references to `odo` instead of odo
- Grammatical changes
- Updates to links / operator hub version

**Which issue(s) this PR fixes:**

N/A

**PR acceptance criteria:**

- [X] Unit test

- [X] Integration test

- [X] Documentation

**How to test changes / Special notes to the reviewer:**

N/A

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: docs/website/docs/getting-started/cluster-setup/kubernetes.md [hunk 1]
```diff
@@ -4,61 +4,64 @@ sidebar_position: 1
 ---
 
 # Setting up a Kubernetes cluster
-*Note that this guide is only helpful in setting up a development environment; this setup is not recommended for a production environment.*
+
+## Introduction
+This guide is helpful in setting up a development environment intended to be used with `odo`; this setup is not recommended for a production environment.
+
+`odo` can be used with ANY Kubernetes cluster. However, this development environment will ensure complete coverage of all features (including advanced) of `odo`.
```
Comment (dharmit) [inline docs/website/docs/getting-started/cluster-setup/kubernetes.md:11]: What's the significance of `including advanced` here? 🤔

File: docs/website/docs/getting-started/cluster-setup/kubernetes.md [hunk 2]
```diff
@@ -4,61 +4,64 @@ sidebar_position: 1
...
 
 ## Prerequisites
-* You have a Kubernetes cluster setup, this could for example be a [minikube](https://minikube.sigs.k8s.io/docs/start/) cluster.
-* You have admin privileges to the cluster, since the Operator installation is only possible with an admin user.
+* You have a Kubernetes cluster set up (such as [minikube](https://minikube.sigs.k8s.io/docs/start/))
+* You have admin privileges to the cluster
```
Comment (dharmit) [inline docs/website/docs/getting-started/cluster-setup/kubernetes.md:15]: Why do you think that the part about Operator installation should be removed? It was added to convey why the admin access is needed.

File: docs/website/docs/getting-started/cluster-setup/kubernetes.md [hunk 3]
```diff
@@ -4,61 +4,64 @@ sidebar_position: 1
...
+
+## Summary
+* An Ingress Controller in order to use `odo url create`
+* Operator Lifecycle Manager in order to use `odo service`
```
Comment (dharmit) [inline docs/website/docs/getting-started/cluster-setup/kubernetes.md:19]:
```suggestion
* Operator Lifecycle Manager in order to use `odo service create`
```

File: docs/website/docs/getting-started/cluster-setup/kubernetes.md [hunk 4]
```diff
@@ -4,61 +4,64 @@ sidebar_position: 1
...
 ## Installing the Operator Lifecycle Manager (OLM)
-The Operator Lifecycle Manager(OLM) is a component of the Operator Framework, an open source toolkit to manage Kubernetes native applications, called Operators, in a streamlined and scalable way. [(Source)](https://olm.operatorframework.io/)
 
-[//]: # (Move this section to Architecture > Service Binding or create a new Operators doc)
-What are Operators?
->The Operator pattern aims to capture the key aim of a human operator who is managing a service or set of services. Human operators who look after specific applications and services have deep knowledge of how the system ought to behave, how to deploy it, and how to react if there are problems.
->
->People who run workloads on Kubernetes often like to use automation to take care of repeatable tasks. The Operator pattern captures how you can write code to automate a task beyond what Kubernetes itself provides.
-> [(Source)](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/#motivation)
+Creaing the Operator Lifecycle Manager (OLM) is required to use the `odo service create` feature.
```
Comment (dharmit) [inline docs/website/docs/getting-started/cluster-setup/kubernetes.md:45]: I don't think we are creating OLM here, we are installing it. WDYT?
```suggestion
Installing the Operator Lifecycle Manager (OLM) is required to use the `odo service create` feature.
```

**Review by dharmit — CHANGES_REQUESTED** [review]

**Comment by dharmit** [comment]
@cdrage can you remove the `$` sign from all the shell commands other than sample outputs? The reason for this is that docusaurus provides a button to copy what's put into code blocks. This shows up when you hover your mouse over the code block - just the way GitHub does it. If we add a `$`, user will not be able to simply copy-paste commands from odo docs without removing the `$` first.

**Comment by cdrage** [comment]
Thanks for the review @dharmit i've pushed your reccomended changes.

Sorry about the `$` it's a habit from the openshift docs days where every command required the use of $ to indicate to run it without sudo / root!

**Comment by dharmit** [comment]
> Sorry about the `$` it's a habit from the openshift docs days where every command required the use of $ to indicate to run it without sudo / root!

No worries. I still find it awkward to write without a `$` in docusaurus when I do it all the time on GH. 😃

**Pull request merged**